From f494bb5cb49014b2248d51709d124e43e5b22fbb Mon Sep 17 00:00:00 2001
From: Zhang Zekun <zhangzekun11@huawei.com>
Date: Tue, 30 Jul 2024 16:06:23 +0800
Subject: [PATCH 882/960] downstream: iommu/arm-smmu-v3: Fix the compile
 warning

Fix the following compile warning:

WARNING: modpost: vmlinux.o(.text+0x854620):
...
The function arm_smmu_device_probe() references
the variable __initdata arm_smmu_v3_plat_info.
This is often because arm_smmu_device_probe lacks a __initdata
annotation or the annotation of arm_smmu_v3_plat_info is wrong.

Fixes: b11eff2baba2 ("downstream: iommu/arm-smmu-v3: Change the style to identify the the hisi_platform")
Signed-off-by: Zhang Zekun <zhangzekun11@huawei.com>
Signed-off-by: Deng Guangxing <dengguangxing@huawei.com>
Signed-off-by: Zhang Guangzhi <zhangguangzhi3@huawei.com>
Signed-off-by: Li Wei <liwei818@huawei.com>
---
 drivers/iommu/arm/arm-smmu-v3/arm-smmu-v3.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/iommu/arm/arm-smmu-v3/arm-smmu-v3.c b/drivers/iommu/arm/arm-smmu-v3/arm-smmu-v3.c
index 61e59cb8333c..7f6676bb6928 100644
--- a/drivers/iommu/arm/arm-smmu-v3/arm-smmu-v3.c
+++ b/drivers/iommu/arm/arm-smmu-v3/arm-smmu-v3.c
@@ -5242,7 +5242,7 @@ static int arm_smmu_device_hw_probe(struct arm_smmu_device *smmu)
 }
 
 #ifdef CONFIG_ACPI
-static struct acpi_platform_list arm_smmu_v3_plat_info[] __initdata = {
+static struct acpi_platform_list arm_smmu_v3_plat_info[] = {
 	/* HiSilicon Hip09 Platform */
 	{"HISI  ", "HIP09   ", 0, ACPI_SIG_IORT, greater_than_or_equal,
 	 "Erratum #162100602", 0},
-- 
2.43.0

